tests/test_actions.py

```diff
@@ -0,0 +1,275 @@
+
+    @patch.object(actions, 'datetime')
+    def test_backup_filename_format(self, mock_datetime):
+        """Test that backups are saved to the file with expected time-stamped name"""
+        freeze_time = datetime(2020, 1, 1)
+        mock_datetime.now = MagicMock(return_value=freeze_time)
+        timestamp = freeze_time.strftime('%Y-%m-%d_%H-%M-%S')
+        expected_path = os.path.join(actions.PKI_BACKUP,
+                                     'easy-rsa-{}.tar.gz'.format(timestamp))
+        expected_format = 'w:gz'
+
+        self.call_action()
+        actions.tarfile.open.assert_called_with(expected_path,
+                                                mode=expected_format)
+        actions.function_fail.assert_not_called()
+
+    @patch.object(actions, 'datetime')
+    def test_response(self, mock_datetime):
+        """Test successful backup response"""
+        freeze_time = datetime(2020, 1, 1)
+        mock_datetime.now = MagicMock(return_value=freeze_time)
+        timestamp = freeze_time.strftime('%Y-%m-%d_%H-%M-%S')
+        expected_path = os.path.join(actions.PKI_BACKUP,
+                                     'easy-rsa-{}.tar.gz'.format(timestamp))
+        local_unit = 'easyrsa/0'
+        actions.local_unit.return_value = local_unit
+        self.call_action()
+        expected_arguments = {
+            'command': 'juju scp {}:{} .'.format(local_unit, expected_path),
+            'message': 'Backup archive created successfully. Use the juju scp'
+                       ' command to copy it to your local machine.'
+        }
+
+        actions.function_set.assert_called_with(expected_arguments)
+        actions.function_fail.assert_not_called()
+
+
+class ListBackupTests(_ActionTestCase):
+
+    NAME = 'list-backups'
+
+    def setUp(self, to_mock=None):
+        additional_mocks = {
+            actions.os: ['listdir'],
+        }
+        super(ListBackupTests, self).setUp(to_mock=additional_mocks)
+
+    def test_missing_backup_dir_ok(self):
+        """Don't fail if backup directory does not exist"""
+        actions.os.listdir.side_effect = FileNotFoundError()
+        self.call_action()
+        actions.function_fail.assert_not_called()
+
+    def test_no_backups_response(self):
+        """Test reposnse if there are no backups"""
+        actions.os.listdir.return_value = []
+        self.call_action()
+
+        expected_response = {'message': 'There are no available backup files.'}
+        actions.function_set.assert_called_with(expected_response)
+        actions.function_fail.assert_not_called()
+
+    def test_backup_list_response(self):
+        """Test response containing list of available backups"""
+        backups = ['backup1.tar.gz', 'backup2.tar.gz']
+        actions.os.listdir.return_value = backups
+        self.call_action()
+
+        expected_text = 'Available backup files:'
+        for file in backups:
+            expected_text += '\n{}'.format(file)
```

Comment:
`expected_text = "Available backup files:\n{}".format("\n".join(backups))`

Reply:
this is much better. thanks :)